src/ashell/jerry-code.c

```diff
@@ -74,6 +120,100 @@ static void javascript_print_value(const jerry_value_t value)
     jerry_port_console("\n");
 }
 
+static int read_file(const char *file_name, char **file_buf)
+{
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    ssize_t size = fs_size(fp);
+    if (size == 0) {
+        comms_printf("[ERR] Empty file (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    *file_buf = (char *)zjs_malloc(size);
+    if (*file_buf == NULL) {
+        comms_printf("[ERR] Not enough memory for (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    ssize_t brw = fs_read(fp, *file_buf, size);
+
+    if (brw != size) {
+        comms_printf("[ERR] Failed loading code %s\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    return size;
+}
+
+static void find_requires(char *filebuf)
+{
+    char *ptr1 = filebuf;
+    char *ptr2 = NULL;
+    char *filestr = NULL;
+    char *ext_check = NULL;
+    while(ptr1 != NULL){
+
+        // Find next instance of "require"
+        ptr1 = strstr(ptr1, "require");
+
+        if (ptr1 != NULL) {
+            // Find the text between the two " " after require
+            ptr1 = strchr(ptr1, '"')+1;
```

Comment:
Yeah I assume  require      (   "test.js");  is valid in javascript, so +7 won't work in that case.